### PR TITLE
Add changelog for the v1.4.0 release

### PR DESCRIPTION
**Pull request by xmudrii**
**What this PR does / why we need it**:

Add the changelog for the v1.4.0 release.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/assign @kron4eg 
cc @Mita-Kubermatic @shaase-ctrl

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,19 +1,114 @@
...
+
+* KubeOne 1.4.0-beta.0 introduces the new KubeOneCluster v1beta2 API
+  * The existing KubeOneCluster v1beta1 manifests can be migrated by using the `kubeone config migrate` command
+  * The `kubeone config print` command now uses the new v1beta2 API
+  * The existing KubeOneCluster v1beta1 API is considered deprecated and will be removed in KubeOne 1.6+
+  * Highlights:
+    * The API group has been changed from `kubeone.io` to `kubeone.k8c.io`
+    * The AssetConfiguration API has been removed from the v1beta2 API. The AssetConfiguration API can still be used with the v1beta1 API, but we highly recommend migrating away because the v1beta1 API is deprecated
+    * The PodPresets feature has been removed from the v1beta2 API because Kubernetes removed support for PodPresets in Kubernetes 1.20
+    * Packet (`packet`) cloud provider has been rebranded to Equinix Metal (`equinixmetal`). The existing Packet cluster will work with `equinixmetal` cloud provider, however, manual migration steps are required if you want to use new Terraform configs for Equinix Metal
```
[the remark this answers is not in the document]
Comment (xmudrii) [reply CHANGELOG.md]: Done, PTAL.

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve
/lgtm

**Pull request merged**